floyd/cli/data.py

```diff
@@ -141,6 +141,66 @@ def clone(id):
                               delete_after_untar=True)
 
 
+@click.command()
+@click.argument('id', nargs=1)
+def listfiles(id):
+    """
+    List files in the given dataset
+    """
+
+    data_source = DataClient().get(normalize_data_name(id, use_data_config=False))
+    if id and not data_source:
+        # Try with the raw ID
+        data_source = DataClient().get(id)
+
+    if not data_source:
+        if 'output' in id:
```

Comment:
you should be able to clone output from a job, just not when it's running. So the message here is not correct. you can change the logic to the following:

```python
if "output" in name:
    # check for job state to make sure it's not queued or running

data_source = DataClient().get(normalize_data_name(id, use_data_config=False))
if not data_source:
    data_source = DataClient().get(name)

if not data_source:
    floyd_logger.info("bla bla bla data not found")
```

Reply:
If checking job state is too much work, you can just exit and print out "cannot list files in job output".

Reply:
If you can clone the output of a job, but not a running job - doesn't "You cannot clone the output of a running job. You need to wait for it to finish" capture that already, or am I missing something? (I'm following the pattern from the `clone` command.)

Reply:
you are only checking to see if it's a job output without actually checking the state right?

Reply:
Right. Are you concerned there could be other reasons that initialising `data_source` doesn't work out, other than trying to get at output from a running job?

Reply:
Oh, sorry, I misread the code. I think this is good for now. It would be nice to still send out the request and actually check the state though.

Reply:
Yeah, I'm also not a huge fan of this way (also because e.g. if initialising `data_source` fails for a different reason it'll just silently exit) - but if it's worked fine for `clone` so far, maybe we can let it be for the time being.